lib/persisted-model.js

```diff
@@ -1527,6 +1546,107 @@ module.exports = function(registry) {
     });
   };
 
+  /**
+   * Subscribe to changes of this model.
+   *
+   * @param {Object} options
+   * @param {Object} options.where Only changes to models matching this where filter will be included in the `ChangeStream`.
+   * @callback {Function} callback
+   * @param {Error} err
+   * @param {ChangeStream} changes
+   */
+
+  PersistedModel.subscribe = function(options, cb) {
+    if(typeof options === 'function') {
+      cb = options;
+      options = undefined;
+    }
+
+    var idName = this.getIdName();
+    var Model = this;
+    var changes = new PassThrough({objectMode: true});
+    var writeable = true;
+    var ref = {changes: changes};
+
+    changes.destroy = function() {
+      changes.removeAllListeners('error');
+      changes.removeAllListeners('end');
+      writeable = false;
+      delete ref.changes;
+    };
+
+    changes.on('error', function() {
+      writeable = false;
+    });
+    changes.on('end', function() {
+      writeable = false;
+    });
+
+    process.nextTick(function() {
+      cb(null, changes);
+    });
+
+    Model.observe('before save', createChangeHandler('save', true));
+    Model.observe('before delete', createChangeHandler('delete', true));
```

Comment:
In my opinion it does not make much sense to send optimistic changes from "before" hooks.
 - When running on the server, the latency of database calls is usually negligible, thus an "optimistic" update does provide much value IMO.
 - It would make sense to emit optimistic changes when running in the browser, but IIRC, the isomorphic client does not invoke operation hooks because the remote connector completely bypasses DAO methods.

I think we need to find a different solution for emitting optimistic updates.